custom/staking/keeper/keeper.go

```diff
@@ -0,0 +1,125 @@
+	// unbonded after the Endblocker (go from Bonded -> Unbonding during
+	// ApplyAndReturnValidatorSetUpdates and then Unbonding -> Unbonded during
+	// UnbondAllMatureValidatorQueue).
+	println("BlockValidatorUpdates Custom Staking Module")
+	params := k.Stakingmiddleware.GetParams(ctx)
+	println("BlocksPerEpoch: ", params.BlocksPerEpoch)
+	should_execute_batch := (hight % int64(params.BlocksPerEpoch)) == 0
+	var validatorUpdates []abcicometbft.ValidatorUpdate
+	if should_execute_batch {
+		println("Should Execute Batch: ", hight)
+		v, err := k.ApplyAndReturnValidatorSetUpdates(ctx)
+		if err != nil {
+			panic(err)
+		}
+		validatorUpdates = v
+	}
+
+	// unbond all mature validators from the unbonding queue
+	k.UnbondAllMatureValidators(ctx)
+
+	// Remove all mature unbonding delegations from the ubd queue.
+	matureUnbonds := k.DequeueAllMatureUBDQueue(ctx, ctx.BlockHeader().Time)
+	for _, dvPair := range matureUnbonds {
+		addr, err := sdk.ValAddressFromBech32(dvPair.ValidatorAddress)
+		if err != nil {
+			panic(err)
+		}
+		delegatorAddress := sdk.MustAccAddressFromBech32(dvPair.DelegatorAddress)
+
+		balances, err := k.CompleteUnbonding(ctx, delegatorAddress, addr)
+		if err != nil {
+			continue
+		}
+
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeCompleteUnbonding,
+				sdk.NewAttribute(sdk.AttributeKeyAmount, balances.String()),
+				sdk.NewAttribute(types.AttributeKeyValidator, dvPair.ValidatorAddress),
+				sdk.NewAttribute(types.AttributeKeyDelegator, dvPair.DelegatorAddress),
+			),
+		)
+	}
+
+	// Remove all mature redelegations from the red queue.
+	matureRedelegations := k.DequeueAllMatureRedelegationQueue(ctx, ctx.BlockHeader().Time)
+	for _, dvvTriplet := range matureRedelegations {
+		valSrcAddr, err := sdk.ValAddressFromBech32(dvvTriplet.ValidatorSrcAddress)
+		if err != nil {
+			panic(err)
+		}
+		valDstAddr, err := sdk.ValAddressFromBech32(dvvTriplet.ValidatorDstAddress)
+		if err != nil {
+			panic(err)
+		}
+		delegatorAddress := sdk.MustAccAddressFromBech32(dvvTriplet.DelegatorAddress)
+
+		balances, err := k.CompleteRedelegation(
+			ctx,
+			delegatorAddress,
+			valSrcAddr,
+			valDstAddr,
+		)
+		if err != nil {
+			continue
+		}
+
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeCompleteRedelegation,
+				sdk.NewAttribute(sdk.AttributeKeyAmount, balances.String()),
+				sdk.NewAttribute(types.AttributeKeyDelegator, dvvTriplet.DelegatorAddress),
+				sdk.NewAttribute(types.AttributeKeySrcValidator, dvvTriplet.ValidatorSrcAddress),
+				sdk.NewAttribute(types.AttributeKeyDstValidator, dvvTriplet.ValidatorDstAddress),
+			),
+		)
+	}
+
+	return validatorUpdates
+}
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	staking stakingkeeper.Keeper,
+	acck accountkeeper.AccountKeeper,
+	stakingmiddleware *stakingmiddleware.Keeper,
+	authority string,
+) Keeper {
+	keeper := Keeper{
+		Keeper:            staking,
+		acck:              acck,
```

Comment:
acck?

Reply:
removed that dependency. thanks. left from prev implementation.